testplan/runners/pools/tasks/base.py

```diff
@@ -184,41 +185,51 @@ def materialize(self, target=None):
                     raise AttributeError
             except AttributeError:
                 if callable(target):
-                    return self.materialize(
-                        target(*self._args, **self._kwargs)
-                    )
+                    inner_target = target(*self._args, **self._kwargs)
+                    if inner_target:
+                        return self.materialize(inner_target)
+                    else:
+                        raise TaskMaterializationError(
+                            errmsg.format(target.__name__)
+                        )
                 try:
                     name = target.__class__.__name__
                 except:
                     name = target
                 raise RuntimeError(
-                    ("Task {} must have a " ".run() method.").format(name)
+                    "Target {} must have a `run` method.".format(name)
                 )
             else:
                 return target
         else:
-            target = self._string_to_target()
-            return self.materialize(target(*self._args, **self._kwargs))
+            target = self._string_to_target()(*self._args, **self._kwargs)
+            if target:
+                return self.materialize(target)
+            else:
+                raise TaskMaterializationError(errmsg.format(self._target))
 
     def _string_to_target(self):
+        """Dynamically load an object from a module by target name."""
         path_inserted = False
-        if isinstance(self._path, six.string_types):
+        if (
+            isinstance(self._path, six.string_types)
+            and self._path not in sys.path
```

Comment:
I think the point here is to insert this path at the very beginning of sys.path even if it 's already in sys.path.

Reply:
OK, I'll recover it back.